### ISSUE TITLE
hGetAll and hMSet doesn't work well on 3.1.0

### ISSUE THREAD

**Issue by fantiq** [phpredis/phpredis#1240]
### I'm seeing this behaviour on
- OS: centos 6.8
- Redis: 3.2.6
- PHP:7.0.9
- phpredis:3.1.0

### Steps to reproduce, backtrace or example script
the script code

##### create a hash table
```
$redis->hMSet("test", [
    "name" => "fantiq",
    "count" => 10
]);
```
##### change the value of the hash table:
```
$result = $redis->hGetAll("test");
$result["count"] = 11;
$redis->hMSet("test", $result);
```

### Expected behaviour

expect get a hash table like this:

|row|key|val|
|--|--|--|
|1|name|fantiq|
|2|count|21|

### Actual behaviour

|row|key|val|
|--|--|--|
|1|name|fantiq|
|2|count|10|
|3|test|11|

>i find this from the phpredis version 3.1.0 . and i try the version 3.0.0 ,the work well . this is only part of the appearance , the is some other strange behaviour .

### I've checked
- [x] There is no similar issue from other users
- [ ] Issue isn't fixed in `develop` branch

**Comment by yatsukhnenko** [comment]
@fantiq, it might be fixed in version >= 3.1.1

**Comment by fantiq** [comment]
@yatsukhnenko thanks
